lib/elixir/lib/calendar/iso.ex

```diff
@@ -289,33 +342,64 @@ defmodule Calendar.ISO do
   """
   @doc since: "1.10.0"
   @impl true
-  def parse_date("-" <> string) when is_binary(string),
-    do: parse_date(string, -1)
+  def parse_date(string) when is_binary(string),
+    do: parse_date(string, :extended)
 
-  def parse_date("+" <> string) when is_binary(string),
-    do: parse_date(string, 1)
+  @doc """
+  Parses a date `string` according to a given `format`.
 
-  def parse_date(string) when is_binary(string),
-    do: parse_date(string, 1)
+  The `format` can either be `:basic` or `:extended`.
 
-  defp parse_date(string, multiplier) do
-    with unquote(match_date) <- string, true <- unquote(guard_date) do
-      {year, month, day} = unquote(read_date)
-      year = multiplier * year
+  For more information on supported strings, see how this
+  module implements [ISO 8601](#module-iso-8601-compliance).
 
-      if valid_date?(year, month, day) do
-        {:ok, {year, month, day}}
-      else
-        {:error, :invalid_date}
-      end
+  ## Examples
+
+      iex> Calendar.ISO.parse_date("20150123", :basic)
+      {:ok, {2015, 1, 23}}
+      iex> Calendar.ISO.parse_date("20150123", :extended)
+      {:error, :invalid_format}
+
+  """
+  @doc since: "1.12.0"
+  def parse_date("-" <> string, format) when is_binary(string),
+    do: do_parse_date(string, -1, format)
```

Comment:
Negative dates are only allowed in the extended format, so we should handle the +/- within :extended. :)

Reply:
Good point!

Reply:
I just revised the spec on this--there are two extended formats in play.

One is the general "extended" (as opposed to "basic") format--ie with separators or not. That's what the `:basic` and `:extended` flags here entail.

The other we have decided to support, to work with a wider range of dates, is the "year extension". But our support for that is independent of basic vs extended separator format, and should work with both.

TL;DR I believe this is correct as writ

Reply:
You are correct. I confused expanded with extended. :)